Matveibos/hometask-6/my_hash_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require_relative './my_hash'
+require 'rspec'
+# rubocop: disable Metrics/BlockLength
+describe MyHash do
+  describe '#[]=' do
+    let!(:my_hash) { MyHash.new }
+    context 'when value does not exist' do
+      it 'return pair key-value' do
+        (my_hash[:key1] = '')
+        expect(my_hash[:key1]).to eq('')
+      end
+    end
+
+    context 'when value exist' do
+      before do
+        my_hash[:key1] = 54
+      end
+
+      it 'replace value' do
+        expect(my_hash[:key1]).to eq(54)
+        expect { my_hash[:key1] = 98 }.to change { my_hash[:key1] }.to(98)
+      end
+    end
+
+    context 'when wrong parameters' do
+      it 'return error' do
+        expect { my_hash[] = 3 }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  describe '#[]' do
+    let(:my_hash) { MyHash.new }
+    before do
+      my_hash[:key] = :value1
+    end
+
+    context 'when key exist' do
+      it 'return value' do
+        expect(my_hash[:key]).to eq(:value1)
+      end
+    end
+
+    context 'when key does not exist' do
+      it 'return nil' do
+        expect(my_hash[:not_exist]).to eq(nil)
```

Comment:
There is a bunch of useful methods in the RSpec to check values. For example you can change this test to:
```suggestion
        expect(my_hash[:not_exist]).to be_nil
```
But you solution is okay, this is just for you to know